kvbc/test/sparse_merkle_storage/db_adapter_unit_test.cpp

```diff
@@ -1189,6 +1200,108 @@ TEST_P(db_adapter_custom_blockchain, delete_genesis_and_verify_stale_index) {
   ASSERT_FALSE(hasProvableStaleIndexKeysSince(adapter.getDb(), 2));
 }
 
+TEST_P(db_adapter_custom_blockchain, delete_genesis_and_verify_stale_key_deletion_with_non_provable_keys) {
+  const Sliver skey1{"key1"};
+  const Sliver skey2{"key2"};
+  concord::kvbc::v2MerkleTree::DBAdapter::NonProvableKeySet nonProvableKeySet = {skey1, skey2};
+  auto adapter = DBAdapter{GetParam()->newEmptyDb(), true, nonProvableKeySet};
+
+  const auto ver1StaleNonProvableKey1 = DBKeyManipulator::genNonProvableDbKey(1, skey1);
+  const auto ver1StaleNonProvableKey2 = DBKeyManipulator::genNonProvableDbKey(1, skey2);
+
+  // Add blocks with overlapping keys. Include an empty block.
+  ASSERT_EQ(adapter.addBlock(SetOfKeyValuePairs{
+
+                std::make_pair(skey1, Sliver{"sk1v1"}), std::make_pair(skey2, Sliver{"sk2v1"})}),
+            1);
+  ASSERT_EQ(adapter.addBlock(SetOfKeyValuePairs{}), 2);
+  ASSERT_EQ(adapter.addBlock(SetOfKeyValuePairs{
+
+                std::make_pair(skey1, Sliver{"sk1v2"}), std::make_pair(skey2, Sliver{"sk2v2"})}),
+            3);
+  ASSERT_EQ(adapter.addBlock(SetOfKeyValuePairs{
+
+                std::make_pair(skey1, Sliver{"sk1v3"}), std::make_pair(skey2, Sliver{"sk2v3"})}),
+            4);
+
+  // Verify that the non-provable key for skey1 at block 1 is available.
+  ASSERT_TRUE(adapter.getDb()->has(ver1StaleNonProvableKey1).isOK());
+
+  // Verify that the non-provable key for skey2 at block 1 is available.
+  ASSERT_TRUE(adapter.getDb()->has(ver1StaleNonProvableKey2).isOK());
+
+  // Make sure the are non-provable keys for blocks 3, 4 before deletion.
+  ASSERT_TRUE(hasNonProvableStaleIndexKeysSince(adapter.getDb(), 3));
```

Comment:
What is the rationale for checking if there are keys for block 1 and block 2 and stale indices for block 3 and 4?

Reply:
I am updating the keys by blocks 3 and 4, so I am checking that the stale keys are actually created.

Reply:
Got it, thanks!